stdlib/public/Platform/musl.modulemap

```diff
@@ -2,110 +2,876 @@
 //
 // This source file is part of the Swift.org open source project
 //
-// Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+// Copyright (c) 2014 - 2023 Apple Inc. and the Swift project authors
 // Licensed under Apache License v2.0 with Runtime Library Exception
 //
 // See https://swift.org/LICENSE.txt for license information
 // See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
 //
 //===----------------------------------------------------------------------===//
 
-module SwiftMusl [system] {
-  header "stdc-predef.h"
+module _musl [system] {
+  config_macros _GNU_SOURCE, _BSD_SOURCE, _XOPEN_SOURCE, _POSIX_SOURCE, _POSIX_C_SOURCE, _LARGEFILE64_SOURCE, _ALL_SOURCE, _DEFAULT_SOURCE, __STRICT_ANSI__
+
   header "features.h"
 
-  // C standard library
+  header "bits/musldefs.h"
+
+  exclude header "bits/alltypes.h"
+
+  // Type declarations live in bits
+  module types {
+    umbrella "bits/types"
+      module * {
+      export *
+    }
+  }
+}
+
+module stdc_assert [system] {
+  config_macros NDEBUG
```

Comment:
`config_macros` can't be used in submodules, only top level modules. Also config macros don't apply to textual headers.